api/v1alpha1/pool_webhook.go

```diff
@@ -37,73 +37,67 @@ func (r *Pool) SetupWebhookWithManager(mgr ctrl.Manager) error {
 var _ webhook.Validator = &Pool{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (r *Pool) ValidateCreate() (admission.Warnings, error) {
-	poollog.Info("validate create", "name", r.Name)
+func (p *Pool) ValidateCreate() (admission.Warnings, error) {
+	poollog.Info("validate create", "name", p.Name)
 	ctx := context.TODO()
 
-	if err := r.validateExtraSpec(); err != nil {
+	if err := p.validateExtraSpec(); err != nil {
 		return nil, apierrors.NewInvalid(schema.GroupKind{Group: GroupVersion.Group, Kind: "Pool"},
-			r.Name,
+			p.Name,
 			field.ErrorList{err},
 		)
 	}
 
-	listOpts := &client.ListOptions{
-		Namespace: r.Namespace,
+	poolImage, err := p.GetImageCR(ctx, c)
+	if err != nil {
+		poollog.Error(err, "cannot fetch Image", "error", err)
+		return nil, nil
 	}
 
-	poolList := &PoolList{}
-	if err := c.List(ctx, poolList, listOpts); err != nil {
-		poollog.Error(err, "cannot fetch Pools", "error", err)
-		return nil, err
+	duplicate, duplicateName, err := p.CheckDuplicate(ctx, c, poolImage)
```

Comment:
`CheckDuplicate` Method was added to check duplicate pools with same imagetag, github-scope, flavor and provider on CR creation